ramlfications/parser/types.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2015 Spotify AB
+
+from __future__ import absolute_import, division, print_function
+import attr
+import copy
+from six import MAXSIZE
+from six import iteritems
+
+from ..errors import UnknownTypeError
+from ..parameters import Content
+
+__type_registry = {}
+
+def type_class(klass):
+    if klass.type is not None:
+        __type_registry[klass.type] = klass
```

Comment:
What happens if there are two of the same class types? Here, it'll just overwrite. But if I'm thinking about future-proofing, would there ever be the case where someone uses this decorator without realizing the name of a class is already defined?